### PR TITLE
feat: add contact

### PR DESCRIPTION
**Pull request by edmonday**
[Basecamp Link](https://3.basecamp.com/3105655/buckets/28313680/todos/5094599057)

- add contact form
- add contact form to add contact button
- navigate to person detail page on contact creation
- positioned add contact button on people list

### CODE REVIEW

**Comment by tataihono** [comment]
@edmonday can you merge this in?

File: src/lib/queries/getContact/getContact.ts [hunk 1]
```diff
@@ -21,6 +21,6 @@ export function getContact(id: string): () => Promise<GetContact> {
     await client.content.get<GetContact>(id, {
       type: 'contact',
       appendProcess: 'all',
-      select: ['firstName', 'lastName', 'phoneNumbers', 'emails']
+      select: ['firstName', 'lastName', 'phoneNumbers', 'emails', 'realms']
```
Comment (tataihono) [inline src/lib/queries/getContact/getContact.ts:24]: is this needed?

File: src/lib/queries/getProcessDefinitions.handlers.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import { rest, RestHandler } from 'msw'
```
Comment (tataihono) [inline src/lib/queries/getProcessDefinitions.handlers.ts:1]: these getProcessDefinition files need to be delete

File: src/pages/PeoplePage/PersonAddPage/PersonAddPage.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { Story, Meta } from '@storybook/react'
```
Comment (tataihono) [inline src/pages/PeoplePage/PersonAddPage/PersonAddPage.stories.tsx:1]: these pages need to be moved to pages/PersonAddPage

**Review by tataihono — CHANGES_REQUESTED** [review]

File: src/components/AddContact/AddContact.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import { fireEvent, render, screen } from '@testing-library/react'
+import { Location } from 'history'
+import { SnackbarProvider } from 'notistack'
+import { QueryClient, QueryClientProvider } from 'react-query'
```
Comment (tataihono) [inline src/components/AddContact/AddContact.spec.tsx:4]:
```suggestion
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
```

File: src/components/AddContact/AddContact.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { Container, Typography } from '@mui/material'
+import { useSnackbar } from 'notistack'
+import { ReactElement } from 'react'
+import { useMutation } from 'react-query'
```
Comment (tataihono) [inline src/components/AddContact/AddContact.tsx:4]:
```suggestion
import { useMutation } from '@tanstack/react-query'
```

File: src/components/AddContactButton/AddContactButton.spec.tsx [hunk 6]
```diff
@@ -1,25 +1,31 @@
 import { fireEvent, render, screen } from '@testing-library/react'
-import { Location } from 'history'
-import { MemoryRouter, Route } from 'react-router'
+import { SnackbarProvider } from 'notistack'
+import { QueryClient, QueryClientProvider } from 'react-query'
```
Comment (tataihono) [inline src/components/AddContactButton/AddContactButton.spec.tsx:3]:
```suggestion
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
```

File: src/components/ContactForm/ContactForm.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+import { QueryClient, QueryClientProvider } from 'react-query'
```
Comment (tataihono) [inline src/components/ContactForm/ContactForm.spec.tsx:2]:
```suggestion
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
```

File: src/lib/mutations/createContact.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import { MutationFunction } from 'react-query'
```
Comment (tataihono) [inline src/lib/mutations/createContact.ts:1]:
```suggestion
import { MutationFunction } from '@tanstack/react-query'
```

**Review by tataihono — CHANGES_REQUESTED** [review]

File: src/@types/fluro.d.ts [hunk 9]
```diff
@@ -14,6 +14,11 @@ declare module 'fluro' {
     title: string
   }
 
+  export interface Realm {
+    _id: string
+    title: string
+  }
+
```
[the remark this answers is not in the document]
Comment (edmonday) [reply src/@types/fluro.d.ts:21]: True, I'll drop this.

**Comment by edmonday** [comment]
@tataihono made the changes after your review

**Comment by edmonday** [comment]
@tataihono  tests have been fixed, just waiting on the pending UI Tests

File: package.json [hunk 10]
```diff
@@ -30,7 +30,6 @@
     "@types/react-router": "^5.1.18",
     "@types/react-router-dom": "^5.3.3",
     "chromatic": "^6.7.0",
-    "date-fns": "^2.28.0",
```
Comment (tataihono) [inline package.json:33]: oh. this should not be removed.

File: src/components/AddContact/AddContact.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
+import { Story, Meta } from '@storybook/react'
+import { userEvent, within, screen } from '@storybook/testing-library'
```
Comment (tataihono) [inline src/components/AddContact/AddContact.stories.tsx:2]:
```suggestion
import { userEvent, within } from '@storybook/testing-library'
```

File: src/components/AddContact/AddContact.tsx [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+      values.lastName = 'Unknown'
+    }
+    const response = await mutation.mutateAsync({ ...values })
+    enqueueSnackbar('Contact Created', { variant: 'success' })
+    history.push(`/people/${response._id}`)
+  }
+  return (
+    <>
+      <ContactForm submitLabel="Add Contact" onSubmit={onSubmit} />
+    </>
```
Comment (tataihono) [inline src/components/AddContact/AddContact.tsx:27]:
```suggestion
      <ContactForm submitLabel="Add Contact" onSubmit={onSubmit} />
```

File: src/components/AddContact/AddContact.spec.tsx [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+import { mswServer } from '../../mocks/mswServer'
+
+import { AddContact } from '.'
+
+describe('AddContact', () => {
+  it('creates a contact', async () => {
+    mswServer.use(getRealmSelectableHandler(), createContactHandler())
+    let testLocation: Location | undefined
+    const client = new QueryClient()
+    render(
```
Comment (tataihono) [inline src/components/AddContact/AddContact.spec.tsx:18]: make use of `renderWithProviders` method so you don't need to provide the query client or the snack bar provider.

File: src/components/AddContactButton/AddContactButton.stories.tsx [hunk 14]
```diff
@@ -7,8 +9,25 @@ const AddContactButtonStory = {
...
+export const Clicked = Template.bind({})
+Clicked.play = async ({ canvasElement }): Promise<void> => {
+  const { getByRole } = within(canvasElement)
+  const element = await getByRole('button', { name: 'add' })
+  await userEvent.click(element)
+}
+
+export const Error = Template.bind({})
+Error.play = async ({ canvasElement }): Promise<void> => {
+  const { getByRole, findByRole } = within(canvasElement)
```
Comment (tataihono) [inline src/components/AddContactButton/AddContactButton.stories.tsx:29]:
```suggestion
  const { getByRole } = within(canvasElement)
```

File: src/components/AddContact/AddContact.tsx [hunk 15]
```diff
@@ -0,0 +1,29 @@
+import { useMutation } from '@tanstack/react-query'
```
Comment (tataihono) [inline src/components/AddContact/AddContact.tsx:1]: can this component be merged into the AddContactButton component?

File: src/components/AddContactButton/AddContactButton.spec.tsx [hunk 16]
```diff
@@ -1,25 +1,31 @@
...
 import { AddContactButton } from '.'
 
 describe('AddContactButton', () => {
-  it('navigates to add contact page', () => {
-    let testLocation: Location | undefined
+  it('dialog shows up', () => {
+    mswServer.use(getRealmSelectableHandler(), createContactHandler())
+
+    const queryClient = new QueryClient()
     render(
```
Comment (tataihono) [inline src/components/AddContactButton/AddContactButton.spec.tsx:17]: renderWithProviders

File: src/components/ContactForm/ContactForm.spec.tsx [hunk 17]
```diff
@@ -0,0 +1,90 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+
+import { getRealmSelectableHandler } from '../../lib/queries/getRealmSelectable/getRealmSelectable.handlers'
+import { mswServer } from '../../mocks/mswServer'
+
+import { ContactForm } from './ContactForm'
+
+describe('AddContact', () => {
```
Comment (tataihono) [inline src/components/ContactForm/ContactForm.spec.tsx:9]:
```suggestion
describe('ContactForm', () => {
```

File: src/components/ContactForm/ContactForm.spec.tsx [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+  it('adds a contact', async () => {
+    mswServer.use(getRealmSelectableHandler())
+    const queryClient = new QueryClient()
+    const onSubmit = jest.fn()
+    render(
```
Comment (tataihono) [inline src/components/ContactForm/ContactForm.spec.tsx:14]: renderWithProviders

File: src/components/ContactForm/ContactForm.spec.tsx [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+      })
+    })
+  })
+
+  it('shows error for required fields', async () => {
+    mswServer.use(getRealmSelectableHandler())
+    const queryClient = new QueryClient()
+    const onSubmit = jest.fn()
+
+    render(
```
Comment (tataihono) [inline src/components/ContactForm/ContactForm.spec.tsx:59]: renderWithProviders

File: src/components/ContactForm/ContactForm.spec.tsx [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+    expect(screen.getByText('Realm is required')).toBeInTheDocument()
+  })
+})
+
+it('shows eror when realm is not selected', async () => {
+  mswServer.use(getRealmSelectableHandler())
+  const queryClient = new QueryClient()
+  const onSubmit = jest.fn()
+
+  render(
```
Comment (tataihono) [inline src/components/ContactForm/ContactForm.spec.tsx:80]: renderWithProviders

File: src/lib/queries/getProcessesQuery.ts [hunk 21]
```diff
@@ -0,0 +1,22 @@
+import { client } from '../fluro'
```
Comment (tataihono) [inline src/lib/queries/getProcessesQuery.ts:1]: drop this file.

**Review by tataihono — CHANGES_REQUESTED** [review]

**Comment by edmonday** [comment]
@tataihono changes have been made.

**Review by tataihono — APPROVED** [review]

**Pull request merged**